### PR TITLE
Q4 V1

### PR DESCRIPTION
**Pull request by NaveedGujral**

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -8,10 +8,26 @@ exports.getAllTopics = (req, res, next) => {
...
+    
+exports.getArticleById = (req, res, next) => {
+    const artId = req.params.article_id
+    selectArticleById(artId)
+    .then((article) => {
+        
+        if(article[0] === undefined) {
+            res.status(404).send({ msg: "article does not exist" })
+        }
+
```
Comment (aloush-dev) [inline controllers/controllers.js]: This should really be handeled in the model.
The controller is only for passing data.

File: models/models.js [hunk 2]
```diff
@@ -8,6 +8,14 @@ exports.selectAllTopics = () => {
     })
 }
 
+exports.selectArticleById = (artId) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = ${artId}`)
+    .then(({ rows }) => {
+        // console.log(rows)
```
Comment (aloush-dev) [inline models/models.js]: remove console logs if you no longer need them.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see another test for a 400 bad request, what if the param you pass is not a valid data type.

Please add the missing test and address the comments I have left and resubmit.

Thanks

File: models/models.js [hunk 3]
```diff
@@ -8,6 +8,14 @@ exports.selectAllTopics = () => {
     })
 }
 
+exports.selectArticleById = (artId) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = ${artId}`)
```
Comment (aloush-dev) [inline models/models.js]: Shouldn't be using template literals here it allows for SQL injection.
Please look into sql parameterized query instead.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see another test for a 400 bad request, what if the param you pass is not a valid data type.

Please add the missing test and address the comments I have left and resubmit.

Thanks

File: __tests__/integration-tests.test.js [hunk 4]
```diff
@@ -42,4 +42,42 @@ describe('GET /api tests', () => {
...
+});
+
+describe('GET /api/articles/:article_id tests', () => {
+    test('should respond with an object with the correct information', () => {
+        return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toMatchObject({
+                article: {
```
Comment (nc-hev-mag) [inline __tests__/integration-tests.test.js:78]: nice!

File: models/models.js [hunk 5]
```diff
@@ -8,6 +8,20 @@ exports.selectAllTopics = () => {
...
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [artId])
+    .then(({ rows }) => {
+        const article = rows
+        if (article.length === 0) {
+            return Promise.reject({
+                status: 404,
+                msg: `article does not exist`
+            })
+        }
+        return article
```
Comment (nc-hev-mag) [inline models/models.js:21]: should ideally send back the object here not the array, just so that you don't need to do so in the controller

**Review by nc-hev-mag — APPROVED** [review]
Looks good Naveed - merge away

**Pull request merged**